hoodie-common/src/main/java/com/uber/hoodie/common/util/ReflectionUtils.java

```diff
@@ -58,4 +46,28 @@
         }
     }
 
+    /**
+     * Instantiate a given class with a generic record payload
+     *
+     * @param recordPayloadClass
+     * @param payloadArgs
+     * @param <T>
+     * @return
+     */
+    public static <T extends HoodieRecordPayload> T loadPayload(String recordPayloadClass,
+                                                                       Object [] payloadArgs,
+                                                                       Class<?> ... constructorArgs) {
```

Comment:
rename to `constructorArgTypes`?  this sounds similar to `payloadArgs` now

Reply:
done